### ISSUE TITLE
Waypoints with key but no value

### ISSUE THREAD

**Issue by Phyks** [pdhoopr/gps-to-gpx#10]
Hi,

In my particular use case, I always have all the keys defined but the value is not always defined. Typically, my waypoints can look like

```
[{ elevation: 10, ... }, { elevation : null, ... }, { elevation: 20, ... }]
```

With the current code and the check for `hasOwnProperty`, the property is always defined, so the `<ele>` tag is always emitted (with empty value).

What about switching the `hasOwnProperty` test to `point[eleKey] !== null && point[eleKey] !== undefined` for example? This would avoid emitting such values.

I could also handle this by pre-filtering my data, not sure what sounds best for you.

Thanks!

**Comment by pdhoopr** [comment]
Good catch! This was an oversight on my part when initially writing the code. I agree that what you've described should be the functionality.

However since I released it like this, I'm hesitant to make this change as a v1 feature. In theory, there could be users who have taken advantage of the existing behavior to output empty tags. In practice, I doubt anyone is actually doing that, but I wouldn't want this to break anyone's code if so.

All that being said, I recently started messing around with a v2 of this library, and I fully intend to implement what you've suggested as part of that. This issue should give me some motivation to get more serious about v2 😃 . I'll leave it open for now and plan to address it at that time.

Thank you!

**Comment by Phyks** [comment]
That works fine for me, I'll do the pre-processing on my side :) Thanks!

Feel free to close this one :)
